grudge/geometry/metrics.py

```diff
@@ -566,15 +566,21 @@ def _signed_face_ones(
 
     signed_face_ones_numpy = actx.to_numpy(signed_ones)
 
+    new_group_arrays = []
     for igrp, grp in enumerate(all_faces_conn.groups):
+        grp_field = signed_face_ones_numpy[igrp]
+        sign_mask = np.ones_like(grp_field)
```

Comment:
I have a hard time remembering why this was written this way (I think those `signed_face_ones` arrays are just created this way so that they have the right shape?), but maybe we can improve it a bit.

How about this?
* Remove the `signed_face_ones` and `signed_face_ones_numpy` things.
* Get the discretization `discr = dcoll.discr_from_dd(dd.with_discr_tag(DISCR_TAG_BASE))`
* For each zip(discr.groups, conn.groups), `mask = np.ones((dgrp.nelements, dgrp.nunit_dofs), dtype=discr.real_dtype)`

Would something like that work? The main idea being that we'll just create everything in numpy and then transfer it over to the array context, so we don't have to worry about how writable the arrays are. 

We might want to tag the end result in the same way that `Discretization.zeros` does too?

Reply:
Thanks for the suggestion! What do you think of 85a0d54? (feel free to push directly to this branch if I did something silly, I am very unfamiliar with this code).

Reply:
Looks good to me! (doesn't look like the test failure is related?)